src/norm/pragmas.nim

```diff
@@ -19,3 +19,80 @@ template ro* {.pragma.}
 
 template unique* {.pragma.}
   ## Mark field as unique.
+
+template fk*(val: typed) {.pragma.}
+  ##[ Mark ``int`` field as foreign key. Foreign keys always references the field ``id`` of ``val``. ``val`` should be a Model.
+  ]##
+runnableExamples:
+  import os
+  import norm/model
+  import norm/sqlite
+  import norm/pragmas
+
+  type
+    Product = ref object of Model
+      name : string
+      price : float
+
+    Consumer = ref object of Model
+      name: string
+      productId {.fk: Product.}: int
+
+  proc newProduct(): Product=
+    Product(name: "", price: 0.0)
+
+  proc newConsumer(name: string = "", productId: int = 0): Consumer=
+    Consumer(name: name, productId: productId)
+
+  let dbName = getTempDir() / "example.db"
```

Comment:
That's a great idea to use temp dirs! I should rewrite the tests to use it too! Thanks!